integration/tests/bft/topicvalidator/orchestrator.go

```diff
@@ -0,0 +1,275 @@
+	// give the network enough time to process the unauthorized messages. This ensures us that the unauthorized messages
+	// were indeed dropped and not unprocessed.
+	numOfAuthorizedEvents = 100
+
+	// numOfUnauthorizedEvents the number of unauthorized events per type to send by the test orchestrator.
+	numOfUnauthorizedEvents = 10
+)
+
+// testOrchestrator represents a simple testOrchestrator that passes through all incoming events.
+type testOrchestrator struct {
+	sync.Mutex
+	t                          *testing.T
+	logger                     zerolog.Logger
+	orchestratorNetwork        insecure.OrchestratorNetwork
+	codec                      network.Codec
+	unauthorizedEventsReceived flow.IdentifierList
+	authorizedEventsReceived   flow.IdentifierList
+	unauthorizedEvents         map[flow.Identifier]*insecure.EgressEvent
+	authorizedEvents           map[flow.Identifier]*insecure.EgressEvent
+	authorizedEventsWg         sync.WaitGroup
+	attackerAN                 flow.Identifier
+	attackerEN                 flow.Identifier
+	victimEN                   flow.Identifier
+	victimVN                   flow.Identifier
+}
+
+var _ insecure.AttackOrchestrator = &testOrchestrator{}
+
+func NewOrchestrator(t *testing.T, logger zerolog.Logger, attackerAN, attackerEN, victimEN, victimVN flow.Identifier) *testOrchestrator {
+	orchestrator := &testOrchestrator{
+		t:                          t,
+		logger:                     logger.With().Str("component", "bft_test_orchestrator").Logger(),
+		codec:                      unittest.NetworkCodec(),
+		unauthorizedEventsReceived: flow.IdentifierList{},
+		authorizedEventsReceived:   flow.IdentifierList{},
+		unauthorizedEvents:         make(map[flow.Identifier]*insecure.EgressEvent),
+		authorizedEvents:           make(map[flow.Identifier]*insecure.EgressEvent),
+		authorizedEventsWg:         sync.WaitGroup{},
+		attackerAN:                 attackerAN,
+		attackerEN:                 attackerEN,
+		victimEN:                   victimEN,
+		victimVN:                   victimVN,
+	}
+
+	orchestrator.initUnauthorizedEvents()
+	orchestrator.initAuthorizedEvents()
+
+	return orchestrator
+}
+
+// HandleEgressEvent implements logic of processing the outgoing (egress) events received from a corrupted node.
+func (o *testOrchestrator) HandleEgressEvent(event *insecure.EgressEvent) error {
+	lg := o.logger.With().
+		Hex("corrupt_origin_id", logging.ID(event.CorruptOriginId)).
+		Str("channel", event.Channel.String()).
+		Str("protocol", event.Protocol.String()).
+		Uint32("target_num", event.TargetNum).
+		Str("target_ids", fmt.Sprintf("%v", event.TargetIds)).
+		Str("flow_protocol_event", fmt.Sprintf("%T", event.FlowProtocolEvent)).Logger()
+
+	err := o.orchestratorNetwork.SendEgress(event)
+	if err != nil {
+		// since this is used for testing, if we encounter any RPC send error, crash the testOrchestrator.
```

Comment:
this isn't accurate anymore

Reply:
Yes, we no longer crash the orchestrator due to network errors. Please remove the comment.

Reply:
https://github.com/onflow/flow-go/pull/3580/commits/852de032b043aec14dd08ff68cf8be7edf64eabd